logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -9,11 +9,25 @@ def initialize()
       @queue = java.util.concurrent.SynchronousQueue.new()
     end
 
+    # Push an object to the queue if the queue is full
+    # it will block until the object can be added to the queue.
+    #
+    # @param [Object] Object to add to the queue
     def push(obj)
       @queue.put(obj)
     end
     alias_method(:<<, :push)
 
+    # Offer an object to the queue, wait for the specified amout of time.
+    # If adding to the queue was successfull it wil return true, false otherwise.
+    #
+    # @param [Object] Object to add to the queue
+    # @param [Integer] Time in milliseconds to wait before giving up
+    # @param [Boolean] True if adding was successfull if not it return false
```

Comment:
is this also @param since it's the return value? I'm not too familiar with the syntax